src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -243,21 +243,106 @@ public Rule TransformRuleVersionOne(RuleVersionOne node)
             return rule;
         }
 
+        private List<FileLocation> CreateFileLocationsFromDictionary(IDictionary<string, string> dictionary, Run run)
+        {
+            List<FileLocation> result = null;
+
+            if (dictionary != null)
+            {
+                result = new List<FileLocation>();
+
+                foreach (string key in dictionary.Keys)
+                {
+                    var fileLocation = new FileLocation
+                    {
+                        Uri = new Uri(key, UriKind.RelativeOrAbsolute)
+                    };
+                    result.Add(fileLocation);
+
+                    if (run != null && !string.IsNullOrWhiteSpace(dictionary[key]))
+                    {
+                        // We have contents, so mention this file in run.files
+                        if (run.Files == null)
+                        {
+                            run.Files = new Dictionary<string, FileData>();
+                        }
+
+                        if (!run.Files.ContainsKey(key))
```

Comment:
Should be:
```
if (!run.Files.Contains(key))
{
    run.Files.Add(key, new FileData());
}

File responseFile = run.Files[key];  // Now guaranteed to exist.
responseFile.Contents = new FileContent
{
    Text = dictionary[key];
};
```

Reply:
Ok ok, I was thinking we wouldn't do anything if it already existed. Changed.

Reply:
It's unlikely, but it might exist with some other properties set. Another thing to consider: With this change, I can see at a glance that the code is correct. Before the change, I have to convince myself that the `file` object _could not have_ existed before. It might be true, but with the changed code, I don't have to think about it.